dfdaemon/proxy/proxy.go

```diff
@@ -330,12 +330,16 @@ func (proxy *Proxy) handleHTTPS(w http.ResponseWriter, r *http.Request) {
 	// We have to wait until the connection is closed
 	wg := sync.WaitGroup{}
 	wg.Add(1)
-	http.Serve(&singleUseListener{&customCloseConn{sConn, wg.Done}}, rp)
+	if err := http.Serve(&singleUseListener{&customCloseConn{sConn, wg.Done}}, rp); err != nil {
+		logrus.Error(err)
+	}
 	wg.Wait()
 }
 
 func copyAndClose(dst io.WriteCloser, src io.ReadCloser) {
```

Comment:
I think `copyAndClose` is a util function which should not include output logs and it should return errors to make the caller determines how to deal with the errors.